Add --retries and --timeout optional arguments to command

felddy opened the pull request:
# <!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description

Add options for `--retries` and `--timeout` to the command.

## 💭 Motivation and Context

Some of the fine folks down in sector 7G were asking for such functionality since they are running many scans via `xargs`.  

## 🧪 Testing

Added parameter validation and unit tests to cover both new options. 
Validated tool usage against remote targets.

## 📷 Screenshots (if appropriate)
NA
## 🚥 Types of Changes

<!--- What types of changes does your code introduce? -->
<!--- Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

<!--- Go over all the following points, and put an `x` in all the
boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask.
We're here to help! -->

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

felddy committed:
Change metadata to reflect production status of code.

felddy committed:
Add schema dependency and remote setuptools.

felddy committed:
Replace setuptools.

felddy committed:
Add ability to pass in retries and timeout arguments.
Change the way validation and type conversion is implemented per the 
docopt documentation examples.

felddy committed:
Bump version

felddy committed:
Add retry and timeout values to debug output

felddy committed:
Update documentation with --help output.

jsf9k commented on src/check_cve/check.py:
```diff
@@ -65,23 +70,51 @@ def is_vulnerable(host):
+            "--retries": And(
+                Use(int),
+                lambda n: 0 <= n,
+                error="--retries must be an integer greater than 0.",
```
Your error string says "greater than 0" but your lambda checks if it is greater than _or equal to_ zero.

jsf9k commented on src/check_cve/check.py:
```diff
@@ -65,23 +70,51 @@ def is_vulnerable(host):
+            "--timeout": And(
+                Use(int),
+                lambda n: 1 <= n,
+                error="--timeout must be a an integer greater than 1.",
```
Your error string says "greater than 1" but your lambda checks if it is greater than _or equal to_ one.

felddy replied:
Good catch.  Things got messed up when I converted my language.  I used to say "positive integers", but then realized that didn't capture zero.  Anyway, updated lambda logic, English error message, and the pytests for them in 8380206. 

Crikey... I just messed it up again.  brb.

felddy replied:
Ok.  All good now: bbff80e
But you should double-check my logic.  I must be hungry.

jsf9k reviewed (changes requested):
I noticed a couple of places where an error string is in mild disagreement with the actual code.  Potent work, otherwise.

dav3r commented on src/check_cve/check.py:
```diff
@@ -46,8 +51,8 @@ def is_vulnerable(host):
 
-    logging.debug("Creating connection pool")
-    pool = urllib3.PoolManager(cert_reqs="CERT_NONE")
+    logging.debug(f"Creating connection pool, retries={retries}, timeout={timeout}")
```
This is minor, but as long as we're logging, might as well be nice and mention that the timeout is in seconds

felddy replied:
Good idea.  I was going to add `seconds` to the message but it was going to force the code to wrap.  Came up with a good solution in: 9d42ffc

dav3r replied:
I like it. 👍

dav3r reviewed (approved):
Very nice! ⚙️

felddy committed:
Change exit code for invalid arguments. 
This will match the exit code that docopt parser returns as well as what 
is documented in the --help.

felddy committed:
Fix lying comment.

felddy committed:
Remove from text error since assert prints it.  Change 254 to 1 as in help.

felddy committed:
Make error messages match logic.

felddy committed:
Add units to timeout log debug message.

felddy committed:
Fix timeout validity check.

dav3r commented on src/check_cve/check.py:
```diff
@@ -88,8 +88,8 @@ def main():
-                lambda n: n > 1,
-                error="--timeout must be a an integer greater than 1.",
+                lambda n: n > 0,
+                error="--timeout must be a an integer greater than 0.",
```
You have a bonus `a ` in your error message:
_--timeout must be ~~a~~ an integer greater than 0._

felddy replied:
Got it. 9bc0ece
thx

felddy committed:
Bump version

felddy committed:
Fix english.

felddy committed:
Replace depricated call with replacement.

felddy committed:
Add mock responses for vuln and non-vuln hosts tests.

felddy committed:
Merge branch 'develop' into improvement/retry-timeout-args
